consensus/spos/bls/subroundEndRound.go

```diff
@@ -572,29 +623,29 @@ func (sr *subroundEndRound) verifyNodesOnAggSigFail() ([]string, error) {
 			return nil, err
 		}
 
-		isSuccessfull := true
-		err = sr.SigningHandler().VerifySignatureShare(uint16(i), sigShare, sr.GetData(), sr.Header.GetEpoch())
+		err = sr.checkGoRoutinesThrottler(ctx)
 		if err != nil {
-			isSuccessfull = false
-
-			err = sr.SetJobDone(pk, SrSignature, false)
-			if err != nil {
-				return nil, err
-			}
+			return nil, err
```

Comment:
on the returns here and at line 623 I think it is possible to panic inside the already running goroutines, as they will no longer have access to the context mutex and invalidPubKeys.

Could you create a test to validate/invalidate this?